.github/ISSUE_TEMPLATE/release_template.md

```diff
@@ -0,0 +1,65 @@
+---
+name: Release Checklist
```

Comment:
When I opened an issue for pip GitHub repo, I noticed their template title and description make it very obvious when a template is designed for maintainers only.  I think it is a good idea so that the majority of users who are not maintainers can ignore the template quickly.

e.g. Change name from "Release Checklist" -> "(Maintainers only) Release Checklist"